exec style entrypoint for docker image

wyattwalter opened the pull request:
Hi! I've been playing around with sncli this weekend a bit locally and it's really useful. I see that Docker support was sort of added on recently and I'm not sure if you're interested in maintaining that type of packaging. I noticed that the Docker image is using a shell form entrypoint which is preventing any arguments from being passed to it. From the [Docker documentation](https://docs.docker.com/engine/reference/builder/#entrypoint):

> The shell form prevents any CMD or run command line arguments from being used

This means that no arguments can be passed in, so all you can do from the image is launch the console GUI. That prevents it from being used in any scripting. With this change you can still launch that GUI (which is still the default with no arguments added), but also pass arguments to use it in scripting.

wyattwalter commented:
👍 

I added some notes, but it's kind of inelegant. There's some nuanced behavior between the GUI console and passing input via stdin that I'm not sure how to capture very well. What do you think of what's there? I don't like it that much, but am struggling a bit to communicate this effectively.

Another thing that maybe could be a separate issue is that you could hook this up in Docker Hub to auto-build on master merge / release so the `docker build . -t sncli` step wouldn't be necessary. The installation could be simplified to just a shell alias, something like:

alias sncli='docker run --rm -it -v /tmp:/tmp -v "$HOME/.sncli/:/root/.sncli/" -v "$HOME/.snclirc:/root/.snclirc" insanum/sncli'

That doesn't fix the TTY vs no TTY thing, though 😄

wyattwalter commented:
I setup the Docker Hub thing just as an example of what that looks like. It's hooked up to my fork and pulling in this repo's README: https://hub.docker.com/r/wyattwalter/sncli

swalladge commented:
@wyattwalter hmm ok, I think I see what you mean. I'm not sure we want to add official support for docker (unless there happens to be major demand for running this via docker for some reason). I think I'd rather see the Dockerfile in the repo as an example, and maybe the docker notes you added moved to the wiki here instead. That way, it's still usable/accessible for those that want it, but it's not something else to maintain officially, if that makes sense.

wyattwalter commented:
Sorry, it's been A Week 😄 

Totally understood! I moved my notes to the wiki on the Tips page: https://github.com/insanum/sncli/wiki/Tips-and-Tricks#docker-usage-tips and updated the README's Docker install instructions. How's that look?

wyattwalter commented:
I wasn't sure if you'd like to see it as a separate wiki page. Kinda fits in with "tips and tricks", though it is off _just_ a bit from the spirit of the rest of that page.

swalladge commented:
Thanks @wyattwalter ! Looks great. :)
